### PR TITLE
Use the installer to change log levels

### PR DESCRIPTION
**Pull request by ekohl**
In various places users are told to manually edit files, but that is considered unsupported. This uses the installer to modify files and restart services where needed.

Sadly some things are not exposed as parameters so they're kept in.

There is also some duplication going on and I'd like some writer's opinion on how to address that. See the inline TODO.

* [x] I am familiar with the [contributing](https://github.com/theforeman/foreman-documentation/blob/master/CONTRIBUTING.md) guidelines.

Please cherry-pick my commits into:

* [x] Foreman 3.4/Katello 4.6
* [x] Foreman 3.3/Katello 4.5
* [ ] Foreman 3.2/Katello 4.4
* [ ] Foreman 3.1/Katello 4.3
* For Foreman 3.0 or older, please create a separate PR.
* We do not accept PRs for Foreman 2.3 or older.

### CODE REVIEW

File: guides/common/modules/proc_increasing-the-logging-levels-to-help-with-debugging.adoc [hunk 1]
```diff
@@ -116,62 +104,29 @@ You can find the log for {Project} in `/var/log/foreman/production.log`.
...
-----
-+
-Note that to see logging from some area, debug logging has to be set.
-. Restart {Project} services:
+. Set the logging level:
 +
 [options="nowrap", subs="+quotes,verbatim,attributes"]
 ----
-# {foreman-maintain} service restart
+{foreman-installer} --foreman-logging-level debug \
```
Comment (adamlazik1) [inline guides/common/modules/proc_increasing-the-logging-levels-to-help-with-debugging.adoc:111]: Should something also be done about
```
 :production:
   :type: file
   :layout: pattern
```
Or is this already included int the file and requires no edits from CLI?

File: guides/common/modules/proc_increasing-the-logging-levels-to-help-with-debugging.adoc [hunk 2]
```diff
@@ -116,62 +104,29 @@ You can find the log for {Project} in `/var/log/foreman/production.log`.
...
+--foreman-loggers ldap:true \
+--foreman-loggers permissions:true \
+--foreman-loggers sql:true
 ----
 
-You can find the complete list of loggers with their default values in `/usr/share/foreman/config/application.rb` in the `Foreman::Logging.add_loggers` command.
+TODO: this is a combination of proc_enabling-individual-loggers.adoc and guides/common/modules/proc_enabling-debug-logging.adoc
```
Comment (adamlazik1) [inline guides/common/modules/proc_increasing-the-logging-levels-to-help-with-debugging.adoc:117]: I would in this case delete the section and include aforementioned modules in the assembly instead. Feel free to also make changes to the modules wrapped in `ifdef`s if they require slight changes in this particular context.

Comment (maximiliankolb) [reply guides/common/modules/proc_increasing-the-logging-levels-to-help-with-debugging.adoc:117]: I'd reword this to "You can find the complete list of loggers with their default values in `/usr/share/foreman/config/application.rb` under `Foreman::Logging.add_loggers`."

**Review by maximiliankolb — COMMENTED** [review]
Thanks for this PR! I made a small suggestion, but I feel like it's ok to refer users to a file on Foreman.

File: guides/common/modules/proc_increasing-the-logging-levels-to-help-with-debugging.adoc (hunk 1 above)
Comment (ekohl) [reply guides/common/modules/proc_increasing-the-logging-levels-to-help-with-debugging.adoc:111]: > Or is this already included in the file and requires no edits from CLI?

The installer does all the file editing and restarting. That's the goal of this PR.

File: guides/common/modules/proc_increasing-the-logging-levels-to-help-with-debugging.adoc (hunk 2 above)
Comment (ekohl) [reply guides/common/modules/proc_increasing-the-logging-levels-to-help-with-debugging.adoc:117]: To be clear, `proc_enabling-individual-loggers.adoc` tells you effectively about `--foreman-loggers` and how to use it while `guides/common/modules/proc_enabling-debug-logging.adoc` tells you to use `--foreman-logging-level debug`. This section is essentially telling the users to use both at the same time.

File: guides/common/modules/proc_increasing-the-logging-levels-to-help-with-debugging.adoc [hunk 3]
```diff
@@ -56,25 +56,13 @@ Edit `/etc/hammer/cli_config.yml` and set the `:log_level:`:
...
-# systemctl restart foreman-proxy
-----
-
-[CAUTION]
-====
-Running the installer will revert this change back.
-====
-
 ifdef::katello,orcharhino,satellite[]
 == Increasing the Logging Level For Candlepin
```
Comment (ekohl) [inline guides/common/modules/proc_increasing-the-logging-levels-to-help-with-debugging.adoc:67]: I opened https://github.com/theforeman/puppet-candlepin/pull/226 to make this configurable via the installer.

Comment (ekohl) [reply guides/common/modules/proc_increasing-the-logging-levels-to-help-with-debugging.adoc:67]: I now realize we can only change it view Hiera since there's no corresponding patch in puppet-katello.

**Comment by ekohl** [comment]
cc @asteflova

**Comment by asteflova** [comment]
I'll be working on another issue in sections on logging in the upcoming weeks, so I'll look into the changes proposed here while I'm at it.

File: guides/common/modules/proc_increasing-the-logging-levels-to-help-with-debugging.adoc (hunk 3 above)
Comment (asteflova) [reply guides/common/modules/proc_increasing-the-logging-levels-to-help-with-debugging.adoc:67]: Looking at the changes in https://github.com/theforeman/puppet-candlepin/pull/226/, it seems that Candlepin logging can now also be configured using `foreman-installer`. Can you please confirm @ekohl?

Also, a vaguely related question: Is there a way to print all available log levels (debug, info, warn, ...) from CLI?

Comment (ekohl) [reply guides/common/modules/proc_increasing-the-logging-levels-to-help-with-debugging.adoc:67]: > Looking at the changes in [theforeman/puppet-candlepin#226](https://github.com/theforeman/puppet-candlepin/pull/226), it seems that Candlepin logging can now also be configured using `foreman-installer`. Can you please confirm @ekohl?

While puppet-candlepin can do it, we don't expose that in the installer today. So you can configure it via custom-hiera.yaml, but we try not do document that since it's not intended for end users.

> Also, a vaguely related question: Is there a way to print all available log levels (debug, info, warn, ...) from CLI?

The [data type](https://github.com/theforeman/puppet-candlepin/blob/master/types/loglevel.pp) should validate the user input and if it fails, I think it should give a description but I don't think there's a way before to get the valid options. It would be a good feature request though.

Comment (ekohl) [reply guides/common/modules/proc_increasing-the-logging-levels-to-help-with-debugging.adoc:67]: And I see I didn't link to https://github.com/theforeman/puppet-katello/pull/473 but that would expose it properly to end users.

The first instruction would be replaced with:
```
{foreman-installer} --katello-candlepin-loggers log4j.logger.org.candlepin:DEBUG
```

The second one:
```
{foreman-installer} \
--katello-candlepin-loggers log4j.logger.org.candlepin:DEBUG \
--katello-candlepin-loggers log4j.logger.org.candlepin.resource.ConsumerResource:WARN \
--katello-candlepin-loggers log4j.logger.org.candlepin.resource.HypervisorResource:WARN
```

**Comment by ekohl** [comment]
Closing per docs team triage.
